enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+
+### API Extensions
+
+- Adds a top-level PlatformOperator cluster-scoped CRD.
+- Adds the Bundle and BundleDeployment OLM 1.x API(s).
+- Adds a validating webhook for the Bundle and BundleDeployment APIs.
+- Adds a CRD validation webhook.
+
+### Risks and Mitigations
+
+- Early phases will introduce multiple v1alpha1 CRDs downstream.
+- Early phases will introduce OLM v1 components that are in alpha stages.
+- No concrete migration plan from legacy to OLM v1.
+- Coordination around external team dependencies.
+- The BundleDeployment API has a large net of permissions that may require upfront RBAC to limit this API to admins.
+
+### Drawbacks
+
+- Implementing this PO mechanism using legacy OLM would provide more immediate business value. That implementation is outlined below in the alternatives section.
+- Early phases will deliver through tech preview guidelines. This leads to little customer feedback, given it precludes production clusters.
+- Early phases may introduce a new OLM 1.x control plane that has little-to-no interaction with the existing, legacy OLM 0.x control plane.
+- Early phases may introduce validating webhooks. These webhooks can impact the cluster's overall health when they're not available.
+- Early phases may introduce validating webhooks. By default, these webhooks will be deployed using an HA configuration, but may not have logic to accommodate SNO cluster environments.
+- Early phases may introduce a dynamic watching implementation for managing the underlying OLM operator resources. This has potential performance implications that need to be measured through benchmarking and/or addressed before any GA phase.
+- Future phase designs may require a large scale rewrite of the OLM component.
+
+## Design Details
+
+### Open Questions [optional]
+
+- Does the console need to be able to provide a UI experience for POs?
+- Does having separate APIs for ClusterOperators and PlatformOperators provide a degraded admin UX?
+- How to handle the maxOpenShiftVersion constraint given it's defined in the CSV resource?
+- Dedicated CatalogSource? Dedicated channel? The top-level PO API contains a catalog configuration?
+- Is it possible to have a dedicated PO index image?
+- Should the existing minKubeVersion and maxOpenShiftVersion fields be unified?
+- How are phase 0 requirements enforced?
+- Does this phase need to enforce cluster invariants?
+- What legacy OLM runtime changes are needed?
```

Comment:
hopefully none, do you have something in mind?

Reply:
Depends on the accepted phase 0 design, but even with the current state of the proposal, we may run into issues where a user attempts to install an operator that was already installed via the PO mechanism. I think we may be in the clear here for cluster-scoped resources, where legacy OLM won't stomp on the PO's CRD (assuming that same CRD is present in an OLM bundle), and instead it would fail the Subscription, but I haven't dived deep through that thought experiment yet.

Reply:
Updated this in the latest set of commits. I added some language that we're looking towards addressing this interaction more in phase 1, but there should be minimal OLM runtime changes during phase 0.